ejb3/src/main/java/org/jboss/as/ejb3/util/EjbValidationsUtil.java

```diff
@@ -124,7 +124,7 @@ public static void verifyEjbClassAndDefaultConstructor(final Constructor<?> ctor
     public static boolean verifyEjbPublicMethodAreNotFinalNorStatic(Method[] methods, String classname) {
         boolean isEjbCompliant = true;
         for (Method method : methods) {
-            if (Modifier.isPublic(method.getModifiers()) && !EjbValidationsUtil.verifyMethodIsNotFinalNorStatic(method, classname))
+            if (Object.class != method.getDeclaringClass() && !EjbValidationsUtil.verifyMethodIsNotFinalNorStatic(method, classname))
                 isEjbCompliant = false;
```

Comment:
Nit: this method should fail-fast by immediately returning false when a final/static method is detected,

Reply:
I think the main purpose of this method is to log a warning for each potentially non-compliant public method. So upon the first match, it will continue to find more.

Reply:
Ok, that makes sense.